src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int AMOUNT = 4;
+    private static final int FIGURE_PARAMETERS = 5;
+    private static final int DEFAULT_FIGURE_PARAMETERS = 10;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    private Random random = new Random();
+
+    public Figure getRandomFigure(Figure figure) {
+        if (getQuantityIndex() == 0) {
+            Square square = new Square(getFigureParameters());
+            square.drawFigure();
+        } else if (getQuantityIndex() == 1) {
+            RightTriangle rightTriangle = new RightTriangle(getFigureParameters(),
+                    getFigureParameters());
+            rightTriangle.drawFigure();
+        } else if (getQuantityIndex() == 2) {
+            IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid(getFigureParameters(),
+                    getFigureParameters(), getFigureParameters());
+            isoscelesTrapezoid.drawFigure();
+        } else if (getQuantityIndex() == 3) {
+            Circle circle = new Circle(getFigureParameters());
+            circle.drawFigure();
+        } else {
+            Rectangle rectangle = new Rectangle(getFigureParameters(), getFigureParameters());
+            rectangle.drawFigure();
+        }
+        return figure;
+    }
+
+    public Figure getDefaultFigure(Figure figure) {
```

Comment:
![image](https://user-images.githubusercontent.com/117547096/204099677-138b83fe-7165-4042-b9de-d1884c7bd8ea.png)

Method getDefaultFigure() should not get parameters. It should create and return new defaultCircle